include/grgsm/misc_utils/tmsi_dumper.h

```diff
@@ -48,11 +48,11 @@ namespace gr {
        * class. gsm::tmsi_dumper::make is the public interface for
        * creating new instances.
        */
-      static sptr make();
+      static sptr make(pmt::pmt_t prepend_string, const std::string &filename, bool print_message=true);
```

Comment:
Instead of adding a print_message flag you could specify `/dev/fd/1` for a filename, and if you want the output in both a file and stdout you could pipe it to `tee`, something like `grgsm_tmsi_dumper --file /dev/fd/1 | tee output.file`.